### PR TITLE
Add protect function

### PR DESCRIPTION
**Pull request by gusty**
Adds a function to create a safe version of potentially unsafe functions.

### CODE REVIEW

File: src/FSharpPlus/Extensions.fs [hunk 1]
```diff
@@ -48,6 +48,12 @@ module Option =
     /// <returns>The resulting option value.</returns>
     let ofResult source = match source with Ok x -> Some x | Error _ -> None
 
+    /// Creates a safe version of the suppiled function, which returns an option<'t> instead of throwing exceptions.
```
Comment (jltrem) [inline src/FSharpPlus/Extensions.fs]: spelling: suppiled should be supplied

Comment (gusty) [reply src/FSharpPlus/Extensions.fs]: good catch, thanks !

File: src/FSharpPlus/Extensions.fs [hunk 2]
```diff
@@ -59,6 +65,12 @@ module Result =
     let inline catch f           = function Ok v      -> Ok v     | Error e                 -> (f: 't->_) e : Result<'v,'e>
     let inline either f g        = function Ok v      -> f v      | Error e                 -> g e
 
+    /// Creates a safe version of the supplied function, which returns a Result<'t,exn> instead of throwing exceptions.
+    let protect f x =
```
Comment (wallymathieu) [inline src/FSharpPlus/Extensions.fs]: I think this makes sense in some cases.

**Comment by adz** [comment]
Thanks for this. Was looking for such a function last week!

**Pull request merged**